tests/test_promise.py

```diff
@@ -63,3 +63,16 @@ def callback():
 
         with pytest.raises(TelegramError, match='Error'):
             promise.result()
+
+    def test_done_callback(self):
```

Comment:
Please also test adding the callback before calling `run()` and also both cases with `done_callback` raising an exception.